sql-plugin/src/main/java/com/nvidia/spark/rapids/InternalRowToColumnarBatchIterator.java

```diff
@@ -208,6 +209,23 @@ public ColumnarBatch next() {
     }
   }
 
+  private HostMemoryBuffer[] getHostBuffersWithRetry(SpillableHostBuffer sdb, SpillableHostBuffer sob) {
+    return RmmRapidsRetryIterator.withRetryNoSplit( () -> {
+      HostMemoryBuffer[] hBufs = new HostMemoryBuffer[]{ null, null };
+      try {
+        hBufs[0] = sdb.getHostBuffer();
+        hBufs[1] = sob.getHostBuffer();
+        return hBufs;
+      } finally {
+        // If the second buffer is null, we must have thrown, so close the first one.
+        if ((hBufs[1] == null) && (hBufs[0] != null)) {
+          hBufs[0].close();
+          hBufs[0] = null;
+        }
+      }
```

Comment:
Do we have to test for the exception implicitly, would this also work?

```suggestion
      HostMemoryBuffer dataBuffer = spillableDataBuffer.getHostBuffer();
      HostMemoryBuffer offsetsBuffer = null;
      try {
        offsetsBuffer = spillableOffestBuffer.getHostBuffer();
      } catch (Throwable t) {
        dataBuffer.close();
      }
      return new HostMemoryBuffer[] {dataBuffer, offsetsBuffer};
```

Reply:
Thanks @gerashegalov.  I have simplified this by just doing a try-with-resources for both buffers and incrementing the refcounts in the body so they are retained if didn't throw.